superset/tasks/schedules.py

```diff
@@ -551,10 +552,17 @@ def schedule_alert_query(  # pylint: disable=unused-argument
 
     if report_type == ScheduleType.alert:
         if is_test_alert and recipients:
-            deliver_alert(schedule, recipients)
+            deliver_alert(schedule.id, schedule.slice.id, schedule.label, recipients)
             return
 
-        if run_alert_query(schedule):
+        if run_alert_query(
+            schedule.id,
```

Comment:
Instead of passing in every param needed from schedule, maybe it's cleaner to only pass in the schedule id, then you can get the schedule object within the function and reference all the params there.